File: 01_Language/01_Functions/python/get_class_methods.py
# coding: utf-8

def get_class_methods(class_name):
    return [method for method in vars(class_name).keys() if callable(getattr(class_name, method))]


if __name__ == '__main__':
    class myclass(object):
        def __init__(self):
            pass

        # method 1
        def _myfunc1(self):
            return True

        def myfunc2(self):
            return True


    # print(myclass.__dict__)
    # print(vars(myclass))

    print(get_class_methods(myclass))

    import hashlib

    print(get_class_methods(hashlib))

File: 01_Language/01_Functions/php/get_class_methods.php
<?php

class myclass {
    // constructor
    function __construct()
    {
        return(true);
    }

    // method 1
    private function myfunc1()
    {
        return(true);
    }

    // method 2
    public function myfunc2()
    {
        return(true);
    }
}

$class_methods = get_class_methods('myclass');
// or
$class_methods = get_class_methods(new myclass());

foreach ($class_methods as $method_name) {
    echo "$method_name\n";
}
